pyked/converters.py

```diff
@@ -0,0 +1,741 @@
+            if units == 'Torr':
+                units = 'torr'
+            quantity = 1.0 * unit_registry(units)
+            try:
+                quantity.to(property_units[field])
+            except pint.DimensionalityError:
+                raise KeyError('Error: units incompatible for property ' + name)
+
+            properties[field] = [' '.join([elem.find('value').text, units])]
+
+        else:
+            raise KeyError('Property ' + name + ' not supported as common property.')
+
+    return properties
+
+
+def get_ignition_type(root):
+    """Gets ignition type and target.
+
+    Args:
+        root (`etree.Element`): Root of ReSpecTh XML file
+
+    Returns:
+        properties (`dict`): Dictionary with ignition type/target information
+    """
+    ignition = {}
+    elem = root.find('ignitionType')
+
+    if elem is None:
+        raise AttributeError('missing ignitionType')
+
+    try:
+        ign_target = elem.attrib['target'].rstrip(';').upper()
+    except KeyError:
+        raise AttributeError('missing ignitionType/target')
+    try:
+        ign_type = elem.attrib['type']
+    except KeyError:
+        raise AttributeError('missing ignitionType/type')
+
+    # ReSpecTh allows multiple ignition targets
+    if len(ign_target.split(';')) > 1:
+        raise NotImplementedError('Multiple ignition targets not supported.')
+
+    # Acceptable ignition targets include pressure, temperature, and species
+    # concentrations
+    if ign_target == 'OHEX':
+        ign_target = 'OH*'
+    elif ign_target == 'CHEX':
+        ign_target = 'CH*'
+
+    if ign_target not in ['P', 'T', 'OH', 'OH*', 'CH*', 'CH']:
+        raise KeyError(ign_target + ' not valid ignition target')
+
+    if ign_type not in ['max', 'd/dt max', '1/2 max', 'min']:
+        raise KeyError(ign_type + ' not valid ignition type')
+
+    if ign_target == 'P':
+        ign_target = 'pressure'
+    elif ign_target == 'T':
+        ign_target = 'temperature'
+
+    ignition['type'] = ign_type
+    ignition['target'] = ign_target
+
+    return ignition
+
+
+def get_datapoints(root):
+    """Parse datapoints with ignition delay from file.
+
+    Args:
+        root (`etree.Element`): Root of ReSpecTh XML file
+
+    Returns:
+        properties (`dict`): Dictionary with ignition delay data
+    """
+    # Shock tube experiment will have one data group, while RCM may have one
+    # or two (one for ignition delay, one for volume-history)
+    dataGroups = root.findall('dataGroup')
+    if not dataGroups:
+        raise AttributeError('Error: at least one dataGroup needed.')
+
+    # all situations will have main experimental data in first dataGroup
+    dataGroup = dataGroups[0]
+    property_id = {}
+    unit_id = {}
+    # get properties of dataGroup
+    for prop in dataGroup.findall('property'):
+        unit_id[prop.attrib['id']] = prop.attrib['units']
+        property_id[prop.attrib['id']] = prop.attrib['name'].replace(' ', '-')
```

Comment:
I mean do something like `temp_prop = prop.attrib['name'].replace(' ', '-')` then check the value of `temp_prop`, and if it passes, then fill the value into `property_id`. What happens now is the `property_id` gets a value put into it, then the value is checked, and it errors out if the value doesn't pass. However, I think (might be wrong) there could be a case where the check is skipped for some reason (perhaps in the future if we add some other code or something, or someone catches that error upstream), and it could end up with invalid values in the `property_id` dictionary. Maybe this isn't a real concern, so if you think I'm off base, just ignore me 😄